### ISSUE TITLE
Question about events handlers in multiplayer session

### ISSUE THREAD

**Issue by Blast-1** [FlightControl-Master/MOOSE#790]
I'm having a question about the events handlers logic with multiplayer mission.

When clients (= the players) are entering a multiplayer server, their computers are going to load the lua mission file. And this file could contains script events that can be triggered when a player hit a specific ground target for example. 

Imagine now if the action associated to this event consists on spawning one plane and we have 30 clients on the server at the same time. Will this trigger be activated for each client and then spawn 30 planes when the event will be triggered? 

I took an example with an event handler but i could had take an example with the scheduler trigger.

Ty for your answers.

**Comment by robgrahamau** [comment]
Not how it works, I use event triggers all the time, the way DCS's Multiplayer works is that the SERVER is the one that runs all the actual scripted events, it has primary control,  while clients recieve the mission file in part because it's needed for Track Replay and to reduce some overhead.

You can see that the Server is the only one that actually has the events fire by simply doing an event based on something like IsAlive or OnHit etc.. it only triggers Once per unit actually it's this once triggers that in part makes the very fact that some of the OnClient triggers that are built into DCS itself are busted such a pain in the rear.

To spawn a plane for all 30 clients you would need to basically iterate the event 30 times in a for loop on the event.

So tldr version.. No client's aren't actually 'actively' running the script.. Server is .

**Comment by thebgpikester** [comment]
Also responded in Discord last night on this. Client-Server runs very differently. All you need to know is **if** every client processed a SPAWN, you would get multiple spawns, one for each connected client and they would all fight for who was boss. Same with EVENT. Every client gets the event himself and it's different timing for him and the server, but only the server is processing and running script in action to those events. So if you do anything onEvent, in scripting, it's only processed on the server.
I've tested MENU, EVENT, SPAWN, SCHEDULER, TASK. Client doesn't need to know any of the scripting file and yes there are interesting ramifications and yes, people have been misguided for years in DCS, but I'm not saying anything else!

**Comment by FlightControl-User** [comment]
@Blast-1 I am closing this one.
